bundles/org.openhab.core/src/main/java/org/eclipse/smarthome/core/util/HexUtils.java

```diff
@@ -46,10 +46,25 @@ private HexUtils() {
      * @return the corresponding hex string
      */
     public static String bytesToHex(byte[] bytes, @Nullable CharSequence delimiter) {
-        return Arrays.stream(toObjects(bytes)).map(b -> {
-            int v = b & 0xFF;
-            return "" + hexArray[v >>> 4] + hexArray[v & 0x0F];
-        }).collect(Collectors.joining(delimiter != null ? delimiter : ""));
+        return bytesToHexInt(bytes, delimiter != null ? delimiter : "");
+    }
+
+    private static String bytesToHexInt(byte[] bytes, CharSequence delimiter) {
+        if (bytes.length == 0) {
+            return "";
+        }
+        final StringBuilder sb = new StringBuilder(bytes.length * 2 + delimiter.length() * (bytes.length - 1));
+        int pos = 0;
+        while (pos < bytes.length - 1) {
+            final byte[] hex = byteToHex(bytes[pos++]);
+            sb.append((char) hex[0]);
+            sb.append((char) hex[1]);
+            sb.append(delimiter);
+        }
+        final byte[] hex = byteToHex(bytes[pos++]);
```

Comment:
It seems unnecessary to increase `pos` here after using it: 

```suggestion
        final byte[] hex = byteToHex(bytes[pos]);
```

Reply:
You are absolutely right, it has been done this way to make it visible that it is the same code as in the while loop except the delimiter.
Sure, we don't need to increment `pos` here, but I don't think that this instruction harm. :wink: